### DISCUSSION TITLE
Custom System messages defined in module overridden in DB

### DISCUSSION THREAD

**Discussion by GaryReckard** [craftcms/cms#7980] [Q&A] [answered]
### Description
I’m hooking in to the SystemMessages::EVENT_REGISTER_MESSAGES event to register my custom message that my module uses. These templates do not show up in the craft_systemmessages table. But if I go to /admin/utilities/system-messages, I see them, and if I choose to edit them there, it adds a row to craft_systemmessages. Craft then uses the DB version, rather than the one I defined in code. 

Two questions:

1. If an admin updated one of these messages, and a later deploy included an update in that message as well, would the code update be ignored, or would it override the updated DB version?
2. Is it possible to disable the ability for an admin to edit these system messages?

Thanks!

**Comment by brandonkelly** [comment, marked as answer]
That’s exactly how system messages are supposed to work. The plugin/module registers them and their _default_ subject/body text, but then the admin is free to override them.

> * If an admin updated one of these messages, and a later deploy included an update in that message as well, would the code update be ignored, or would it override the updated DB version?

The change would be ignored, since it’s just a change to the _default_ message text, which the admin has chosen to override with a custom message.

> * Is it possible to disable the ability for an admin to edit these system messages?

No; if you want that then you shouldn’t be using the system message feature in the first place. Just compose the message and set the subject/body text on it manually.

```php
Craft::$app->mailer->compose()
  ->setSubject('Subject Text')
  ->setTextBody('Body text')
  ->setTo('email@address.com')
  ->send();
```

**Reply by GaryReckard** [reply]
Got it, thanks for the explanation, Brandon! 

Also, thanks for moving to Discussions as well! I knew it didn't fit as a 'bug', but somehow overlooked the whole "Discussions" section here.
